### ISSUE TITLE
Support specifying filterwarnings per Python version

### ISSUE THREAD

**Issue by fofoni** [pytest-dev/pytest#11101]
#### What's the problem this feature will solve?

TL;DR: I want this to work:

```toml
# pyprojectl.toml
[tool.pytest.ini_options]
filterwarnings = [
    'ignore::EncodingWarning',  # ignore all EncodingWarnings...
    'default::EncodingWarning:my_package\b',  # ...except those from my_package
]
```

It *does* work when I test `my_package` under Python>=3.10, but I also need to test `my_package` under Python<3.10, and then it breaks with `AttributeError: module 'builtins' has no attribute 'EncodingWarning'`.

Longer description:

AFAIK, it's currently not possible to mention `EncodingWarning` (which was added in Python 3.10) in the list of warning filters in a way that is backwards compatible, and simultaneously use the feature of using regexes to match the *start* module qualified name in the filter. Pytest supports using pyproject.toml as a configuration file, but this file is Python-version-agnostic, so it should contain configuration that works simultaneously for all Python versions. (AFAIK, all other config file formats for pytest are Python-version-agnostic.) This means that since `my_package` supports Python 3.9, I can't mention `EncodingWarning` in pyproject.toml. However, AFAIK, configuring the warning filters list via a configuration file is the only way to enable the regex-matching of module qualnames; specifying the filters via `PYTHONWARNINGS` or `-W` doesn't work. I also tried manually adding filters via `warnings.filterwarnings` conditioned to `sys.version_info>=(3,10)` in `conftest.py`, but this apparently has no effects over my tests' warnings.

To summarize, the following alternatives don't work:

- Don't specify filterwarnings in the config file, and pass customized `-W` flags to the pytest command-lines for each Python version → matching of module names with `-W` flags is performed by comparing the raw string in the filter to the fully qualified module name. I don't think it's reasonable to specify repeated `-W` flags for each of `my_package`'s 15 sub-modules.
- Same, but use `PYTHONWARNINGS` instead of `-W` → same problem; `PYTHONWARNINGS` disables the feature of matching the start of the module qualified name.
- Specify warning filters dinamically in `conftest.py` → no effect.

#### Describe the solution you'd like

It would be nice to have a way of appending strings to the start and/or end of the filterwarnings list conditioned to the Python version. Something like:

```toml
[tool.pytest.ini_options.py310]
filterwarnings_append = ['ignore::EncodingWarning', 'default::EncodingWarning:my_package\b']
```

#### Alternative Solutions

I searched the Internet and asked around in IRC. My research led to the alternatives mentioned above, which are not satisfying (for the reasons mentioned above).
